packages/editor/src/components/rich-text/format-toolbar/link-container.js

```diff
@@ -33,17 +33,29 @@ function getLinkAttributesFromFormat( { attributes: { href = '', target } = {} }
 	return { href, target };
 }
 
-function createLinkFormat( { href, opensInNewWindow } ) {
+/**
+ * Generates the format object that will be applied to the link text.
+ *
+ * @param {string}  href             The href of the link.
+ * @param {boolean} opensInNewWindow Whether this link will open in a new window.
+ * @param {Object}  record           The object that contains the text being wrapped in the a tag.
+ *
+ * @return {Object} The final format object.
+ */
+function createLinkFormat( { href, opensInNewWindow, record } ) {
 	const format = {
 		type: 'a',
 		attributes: {
 			href,
 		},
 	};
 
+	const text = record.text || '';
+
 	if ( opensInNewWindow ) {
 		format.attributes.target = '_blank';
 		format.attributes.rel = 'noreferrer noopener';
+		format.attributes[ 'aria-label' ] = `${ text } (${ __( 'opens in a new tab' ) })`;
```

Comment:
This is still problematic from an i18n standpoint. Should be something like:

```js
// translators: accessibility label for external links, where the argument is the link text
const label = sprintf( __( '%s (opens in a new tab)' ), text );

format.attributes[ 'aria-label' ] = label;
```

Also note that an empty `text` leads to a label that starts with whitespace. Could be handled by calling `String#trim`: `sprintf( … ).trim()`.